app.py

```diff
@@ -307,30 +308,41 @@ def mustachify(original_image_buf):
 def index():
     return render_template(
         'home.html',
+        origin=request.args.get('origin') or '',
     )
 
 
 @app.route('/result/new', methods=["POST"])
 def submit_form():
     file = request.files.get("original")
+    origin = request.form.get('origin')
+    query_params = ''
+    if origin:
+        query_params = '?{}'.format(
+            urllib.parse.urlencode(
```

Comment:
Instead of `urlencode`ing here I think you can just pass in `origin=origin` to `url_for()` and it will append it as a query arg to the returned URL.